router.go

```diff
@@ -284,10 +289,14 @@ func (r *Router) pathChanged(path string, initial bool) {
 //   Route 2: /todos/{category}
 // And the path argument is "/todos/work", the bestRoute would be todos/work
 // because the string "work" matches the literal in Route 1.
-func (r Router) findBestRoute(path string) (bestRoute *route, tokens []string) {
+func (r Router) findBestRoute(path string) (bestRoute *route, tokens []string, params map[string][]string) {
+	parts := strings.Split(path, "?")
+	if len(parts) > 2 {
+		log.Println("Path has more than two parts: " + path)
```

Comment:
I'd rather return an error than panic here.

In this particular case, we're dealing with user input. After all, an end user can type in anything they want in the URL bar. Panicking here would mean we remove the ability of the programmer to easily recover from such a scenario (e.g., by displaying an error message or 404 page).